### PR TITLE
Clean up types

### PR DESCRIPTION
**Pull request by joshbwlng**
Switch from internally defined Context to LogContext.
Clean up other types and how they are imported.
Stop deep importing lodash.

Change-type: major
Signed-off-by: Josh Bowling <josh@balena.io>

---

Calling this a `major` change as we're technically changing the type of the `schema` argument passed to `evaluatePatch()`.

Tested beta package locally in `jellyfish-worker` and all tests passed:
```
Test Suites: 7 passed, 7 total
Tests:       112 passed, 112 total
Snapshots:   0 total
Time:        98.874 s
Ran all test suites matching /.\/test\/integration/i.
```

### CODE REVIEW

**Comment by Ereski** [comment]
@balena-ci rebase

File: lib/index.ts [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
-import staticEval from 'static-eval';
 import * as esprima from 'esprima';
-import * as assert from '@balena/jellyfish-assert';
-import * as jsonpatch from 'fast-json-patch';
-import * as card from './card';
-import type { JSONSchema } from './types';
-import _, { Dictionary } from 'lodash';
+import * as ESTree from 'estree';
+import { applyPatch, compare } from 'fast-json-patch';
+import type { Operation } from 'fast-json-patch';
```
Comment (Ereski) [inline lib/index.ts:15]: Can be merged with the import above

File: lib/index.ts [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+import type { JSONSchema7Object } from 'json-schema';
+import _ from 'lodash';
+import type { Dictionary } from 'lodash';
 import * as objectDeepSearch from 'object-deep-search';
+import staticEval from 'static-eval';
+import type { FormulaPath } from './card';
+import { getFormulasPaths } from './card';
```
Comment (Ereski) [inline lib/index.ts:22]: Can be merged with the import above

**Review by Ereski — COMMENTED** [review]
Cool that you also updated to `jellyfish-types v2` :v:

**Comment by joshbwlng** [comment]
@balena-ci rebase

**Comment by joshbwlng** [comment]
@balena-ci rebase

**Comment by joshbwlng** [comment]
@balena-ci I self certify!

**Pull request merged**